pkg/providers/v1/aws_ec2.go

```diff
@@ -30,6 +30,16 @@ type awsSdkEC2 struct {
 	ec2 ec2iface.EC2API
 }
 
+func (s *awsSdkEC2) DescribeInstanceTopology(request *ec2.DescribeInstanceTopologyInput) ([]*ec2.InstanceTopology, error) {
+	resp, err := s.ec2.DescribeInstanceTopology(request)
+	if err != nil {
+		return nil, fmt.Errorf("error describe AWS Instance Topology: %q", err)
+	} else if len(resp.Instances) == 0 {
+		return nil, nil
```

Comment:
is this branch necessary? wouldn't it be better to have an empty array as a default with no error?

Reply:
changed to `return []*ec2.InstanceTopology{}, nil`